internal/storage/rocksdb/store.go

```diff
@@ -515,7 +592,17 @@ func (rdb *rocksDB) getSingleKey(ro *gorocksdb.ReadOptions, key []byte) ([]*serv
 	defer value.Free()
 	val := toByteArray(value)
 	if val != nil && len(val) > 0 {
-		return []*serverpb.KVPair{&serverpb.KVPair{Key: key, Value: val}}, nil
+		//check ttl.
+		expireTs, _ := getExpireTs(val)
+		if expireTs > 0 && expireTs < time.Now().Unix() {
```

Comment:
Additionally, if the current node is experiencing clock skews w.r.t. to either the master or RAFT leader, then the dependence on `time.Now()` might produce different results and breaks linearizability. Please check.

Reply:
2 Options as discussed

- [ ] We use HLC clock sync between the members.
- [ ] We modify the Get API and allow user to pass currentTS in API. This moves the TS sync logic to client side, but in case of clock skews on client side, issue of linearizability on client side  remains across its nodes.